Add NusaTranslation Bitext Mining (MINERS)

gentaiscool opened the pull request:
NusaTranslation is a parallel dataset for machine translation on 11 Indonesia languages and English.

## Checklist
<!-- Please do not delete this -->

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 


### Adding datasets checklist
<!-- see also https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md -->

**Reason for dataset addition**: ... <!-- Add reason for adding dataset here. E.g. it covers task/language/domain previously not covered -->

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

gentaiscool commented:
@KennethEnevoldsen I created a new PR for NusaTranslation.

gentaiscool commented:
I have completed the checklist.

KennethEnevoldsen commented on mteb/tasks/BitextMining/multilingual/NusaTranslationBitextMining.py:
```diff
@@ -0,0 +1,57 @@
+        }
+
+        """,
+        n_samples={"train": 50200},
```
Should we consider reducing this number? (we have used about 2048 pr lang. previously)

gentaiscool replied:
This is exactly the number of samples I used in MINERS benchmark. It might be comparable if we reduce the number of samples. But, if we want to make a smaller version of the dataset, I can reduce the size if needed. Let me know what you think.

KennethEnevoldsen replied:
Let us keep it as is for now then

KennethEnevoldsen commented on mteb/tasks/BitextMining/multilingual/NusaTranslationBitextMining.py:
```diff
@@ -0,0 +1,57 @@
+
+        """,
+        n_samples={"train": 50200},
+        avg_character_length={"train": 5.62},
```
seems quite short?

gentaiscool replied:
Sorry I put the avg number chars per word. Let me recalculate the number.

KennethEnevoldsen reviewed (commented):
Will you add points and push model results as well?

gentaiscool commented:
Thank you for the review. I addressed them. I updated the avg number of character per sample, added the results, and points. I calculated the points and only consider the languages that are not found in https://github.com/embeddings-benchmark/mteb/pull/910, so there are 6 new languages `abs-Latn`, `bew-Latn`, `bhp-Latn`, `mak-Latn`, `mui-Latn`, `rej-Latn` (2 + 6x4 = 26). Plus, I added the point for the reviewer. @KennethEnevoldsen

KennethEnevoldsen commented:
> I calculated the points and only consider the languages that are not found in https://github.com/embeddings-benchmark/mteb/pull/910, so there are 6 new languages abs-Latn, bew-Latn, bhp-Latn, mak-Latn, mui-Latn, rej-Latn (2 + 6x4 = 26). Plus, I added the point for the reviewer. @KennethEnevoldsen

This is exactly correct
